assets/js/googlesitekit/datastore/site/registry-key.test.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * core/site data store: registryKey tests.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import {
+	createTestRegistry,
+	unsubscribeFromAll,
+} from 'tests/js/utils';
+import { STORE_NAME } from './constants';
+
+describe( 'core/site registryKey', () => {
+	let registry;
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+	} );
+
+	afterEach( () => {
+		unsubscribeFromAll( registry );
+	} );
+
+	describe( 'actions', () => {
+		describe( 'setRegistryKey', () => {
+			it( 'requires the registryKey param', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).setRegistryKey();
+				} ).toThrow( 'registryKey is required.' );
+			} );
+		} );
+	} );
+
+	describe( 'selectors', () => {
+		it( 'generates a registryKey if undefined', async () => {
+			const firstCall = registry.select( STORE_NAME ).getRegistryKey();
+			expect( firstCall ).not.toBeUndefined();
+			expect( registry.select( STORE_NAME ).getRegistryKey() ).toEqual( firstCall );
```

Comment:
Hmmm, this test should be failing if it _is_ generating a `registryKey`. I expect that `firstCall` here should be `undefined`, because we haven't yet set a registry key, and I'd expect we'd need to wait for `hasFinishedResolution('getRegistryKey')` then call this selector again.

```suggestion
			const firstCall = registry.select( STORE_NAME ).getRegistryKey();
			expect( firstCall ).toBeUndefined();

			await subscribeUntil( registry, () => registry
				.select( STORE_NAME )
				.hasFinishedResolution( 'getRegistryKey' )
			);

			expect( registry.select( STORE_NAME ).getRegistryKey() ).not.toBeUndefined();
```

I would expect this test to work.

Reply:
@tofumatt I've been wondering about this for a while. Maybe `@wordpress/data` has some logic built into it that somehow detects what the resolver does in some way? Maybe it'll fulfill immediately if there's no control involved (since it's actually not async then)? I mean this would actually be quite clever.

I think I had noticed this in a few other cases, and the current test passes (basically this selector never returns `undefined` for now). I think we should still write our logic defensively, supporting the case that this returns `undefined`, since we're not sure about the `@wordpress/data` internals yet, but maybe this is how they do it.